README.md

```diff
@@ -6,28 +6,124 @@ See NUbook for a more detailed documentation of this project: <https://nubook.ne
+### Field Grass
 
-To build only the goal, ball or field, run `blender --python <script>` where `<script>` is either `goal`, `ball` or `field` from the `pbr` directory.
+Custom field grass textures to be considered for selection when generating the scene can be placed in the grass directory (by default `resources/grass`).
 
-## Custom Resources
-The resources used for texturing the ball, field and environment by default are found in [`resources/ball_uv`](./resources/ball_uv), [`resources/field_uv`](./resources/field_uv) and [`resources/scene_hdr`](./resources/scene_hdr) respectively.  The directory to look for these resources can be modified in [`scene_config.py`](./pbr/config/scene_config.py) under `field_uv['uv_path']`, `ball['uv_path']` and `scene_hdr['path']`.
+Each grass asset should be placed in a sub directory with the corresponding bump, diffuse, and normal files. For example:
 
-### Field
-A new field UV map and construction can be configured in [`scene_config.py`](./pbr/config/scene_config.py) (by default saved in `<resources_dir>/field_uv`]).
+- `resources/grass/grass_001/grass_001_bump.jpg` (name must include `bump`)
+- `resources/grass/grass_001/grass_001_diffuse.jpg` (name must include `diffuse`)
+- `resources/grass/grass_001/grass_001_normal.jpg` (name must include `normal`)
+
+The `resources.zip` file described in the [Set Up](#set-up) section above has a sample grass texture.
 
 ### Ball
-Custom UV maps to be considered for selection when generating the scene can be placed in the ball UV directory (by default `<resources>/balls`). 
+
+Custom UV maps to be considered for selection when generating the scene can be placed in the ball UV directory (by default `resources/balls`).
+
+Each ball asset should be placed in a sub directory with the corresponding color, mesh, and normal files. For example:
+
+- `resources/balls/ball_001/ball_001_color.png` (name must include `color` or `colour`)
+- `resources/balls/ball_001/ball_001_mesh.fbx` (extension must match the mesh file types configured at `ball["mesh_types"]`)
+- `resources/balls/ball_001/ball_001_normal.png` (name must include `normal`)
+
+The `resources.zip` file described in the [Set Up](#set-up) section above has a sample ball texture.
 
 ### Environment
-Similarly to the ball UV maps, a random HDRI environment image is selected from the pool of images within the scene HDR directory (by default `<resources>/scene_hdr`).
+
+Similarly to the ball UV maps, a random HDRI environment image is selected from the pool of images within the scene HDR directory (by default `resources/hdr`).
+
+Each HDRI image should be placed in a sub directory with the corresponding JSON metadata, mask, and raw HDRI files. For example:
+
+- `resources/hdr/hdr_001/001.json` (must match the metadata file type configured at `environment["info_type"]`)
+- `resources/hdr/hdr_001/001_mask.png` (must match the mask file types configured at `environment["mask_types"]`)
```

Comment:
I think the mask is optional? And only required if `field` is `False` in the `to_draw` map in the JSON file?

Reply:
The mask is used in [`util.points_on_field()`](https://github.com/NUbots/NUpbr/blob/master/pbr/util.py#L232) from [here in the main pbr loop](https://github.com/NUbots/NUpbr/blob/master/pbr/pbr.py#L143-L145).

`point_on_field()` throws if it can't load the mask file, and the call to it doesn't seem to be conditional on `field` being set to `False` in `to_draw`.

So it's probably required?

Reply:
Just tried it to be sure and if `to_draw.field` is `false` then the `cv2.imread(mask_path)` call seemingly creates a blank image and doesn't actually throw. So it does work without a mask file.

More details:

| <code>to_draw["field"]</code> | has mask file | output                              |
|-------------------------------|---------------|-------------------------------------|
| <code>false</code>            | no            | no field in segmentation image      |
| <code>false</code>            | yes           | field appears in segmentation image |
| <code>true</code>             | no            | field appears in segmentation image |
| <code>true</code>             | yes           | field appears in segmentation image |